File: doc-examples/example-java/src/main/java/example/YamlReportService.java
package example;

import io.micronaut.serde.ObjectMapper;
import io.micronaut.serde.yaml.YamlObjectMapper;
import jakarta.inject.Named;
import jakarta.inject.Singleton;

@Singleton
final class YamlReportService {
    private final ObjectMapper yamlMapper;

    YamlReportService(@Named(YamlObjectMapper.YAML) ObjectMapper yamlMapper) {
        this.yamlMapper = yamlMapper;
    }

    ObjectMapper yamlMapper() {
        return yamlMapper;
    }
}

File: doc-examples/example-kotlin/src/main/kotlin/example/YamlReportService.kt
package example

import io.micronaut.serde.ObjectMapper
import io.micronaut.serde.yaml.YamlObjectMapper
import jakarta.inject.Named
import jakarta.inject.Singleton

@Singleton
class YamlReportService(
    @Named(YamlObjectMapper.YAML) private val yamlMapper: ObjectMapper
) {
    fun yamlMapper(): ObjectMapper = yamlMapper
}
